### ISSUE TITLE
Missing ParameterizedType for Option in LTS 3.3

### ISSUE THREAD

**Issue by jchyb** [scala/scala3#20263]
## Compiler version
3.3.0, 3.3.1, 3.3.3 (works correctly in 3.4.0 and 3.4.1)

## Minimized code

```Scala
class ClassWithOption {
  val a: Option[Int] = None
}

object Test {
  def main(args: Array[String]) =
    classOf[ClassWithOption].getDeclaredFields.foreach { field =>
      println(field.toString + " parametrised " + field.getGenericType.isInstanceOf[java.lang.reflect.ParameterizedType])
    }
}
```

## Output
In LTS:
```scala
private final scala.Option ClassWithOption.a parametrised false
```
In 3.4.0 and 3.4.1:
```scala
private final scala.Option ClassWithOption.a parametrised true
```
When using the CFR decompiler on ClassWIthOption I get:
In LTS:
```java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  scala.None$
 *  scala.Option
 */
import scala.None$;
import scala.Option;

public class ClassWithOption {
    private final Option a = None$.MODULE$;

    public Option<Object> a() {
        return this.a;
    }
}

```
In 3.4:
```java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  scala.None$
 *  scala.Option
 */
import scala.None$;
import scala.Option;

public class ClassWithOption {
    private final Option<Object> a = None$.MODULE$;

    public Option<Object> a() {
        return this.a;
    }
}

```

## Expectation
I expect the behavior in 3.4.
Some libraries may use java reflection features which depend on the erased type parameter being there. I suspect overeager type parameter erasure was not intentional, which is why it came back in 3.4.

**Comment by SolalPirelli** [comment]
@Gedochao this issue is LTS-only, if you're interested in backporting the fix to this and #25070, I created https://github.com/dotty-staging/dotty/tree/solal/parameterized-lts which seems to work. Otherwise I guess we could close this issue.

**Comment by Gedochao** [comment]
@tgodzik want to try it?

**Comment by tgodzik** [comment]
So we're talking about porting https://github.com/dotty-staging/dotty/commit/a25c2630ec2dc89356b88922b1a9582a0811b7f2 and it wasn't needed in the latest versions?

**Comment by Gedochao** [comment]
(GitHub has been bugging on me, sorry for the double comment 🙃)

**Comment by SolalPirelli** [comment]
@tgodzik yes, with the rest of the branch if you want a regression test; this specific issue is already fixed in main, but the related #20270 isn't, that's #25543 for main. I figured I might as well try to port the entire logic to 3.3 since I had a little time.

**Comment by tgodzik** [comment]
Added a PR to LTS branch in https://github.com/scala/scala3-lts/pull/862 Thanks!

**Comment by tgodzik** [comment]
Looks like there are some tests failing https://github.com/scala/scala3-lts/actions/runs/23902370550/job/69702066118?pr=862

Any idea why @SolalPirelli ?

**Comment by SolalPirelli** [comment]
No idea, this test didn't cause me any trouble in the original PR, sorry

**Comment by tgodzik** [comment]
Got an additional fix suggested by LLM, looks ok, but any chance to confirm it https://github.com/scala/scala3-lts/pull/862/changes/97b0b6d762ed0a6a13f5434bc7927a7365ddb4f5 anyone?

I will run full community build anyway
